src/s3contents.ts

```diff
@@ -430,13 +429,15 @@ export class Drive implements Contents.IDrive {
 
     await checkS3Object(this._s3Client, this._name, this._root, newLocalPath)
       .then(async () => {
-        console.log('File name already exists!');
+        console.log('Name already exists, constructing new name for object.');
```

Comment:
This function shouldn't be happening in `.then(...)` because it can be done inline with the `await checkS3Object` call.

Reply:
Are you referring to the `incrementName` function call? It's placed in `.then(...)`, as it should only happen if the promise succeeds, which means the object name already exists and a new incremented name should be constructed, otherwise the name should stay the same.

Reply:
Latest commit addresses this.